addons/account/models/account_journal.py

```diff
@@ -866,22 +866,28 @@ def _create_document_from_attachment(self, attachment_ids):
         # As we are coming from the journal, we assume that each attachments
         # will create an invoice with a tentative to enhance with EDI / OCR..
         all_invoices = self.env['account.move']
+        Move = self.env['account.move'].with_company(self.company_id)
```

Comment:
You're right.
*It was a workaround for another problem, but now I solved that so this is useless*

Reply:
No... actually the second move geets created with `company_1_data` even if the journal is on `company_2_data` because the `get_edi_creation` is called on `current_invoice` which takes the `self.env.company`'s  defaults. 
So what I did here was actually necessary :| ...? Halp?

![image](https://github.com/odoo/odoo/assets/1665365/02e08631-81a4-4bb7-8cd1-2ed750735cb7)

Reply:
Done.